x/concentrated-liquidity/math/math.go

```diff
@@ -18,6 +18,7 @@ func Liquidity0(amount osmomath.Int, sqrtPriceA, sqrtPriceB osmomath.BigDec) osm
 
 	// We convert to BigDec to avoid precision loss when calculating liquidity. Without doing this,
 	// our liquidity calculations will be off from our theoretical calculations within our tests.
+	// TODO (perf): consider better conversion helpers to minimize reallocations.
 	amountBigDec := osmomath.BigDecFromDec(amount.ToLegacyDec())
```

Comment:
Note to reviewer:

tracking these here: https://github.com/osmosis-labs/osmosis/issues/6370